src/crypto/openssl/base64.h

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ds/logger.h"
+
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <string>
+#include <vector>
+
+// Inspired by openssl/test/evp_test.c
+// Ref: https://www.openssl.org/docs/man1.1.1/man3/EVP_DecodeBlock.html
+
+namespace crypto
+{
+  struct Base64_openssl
+  {
+    // Decode Base64 into byte stream
+    static std::vector<uint8_t> raw_from_b64(const std::string_view& b64_string)
+    {
+      const auto data = reinterpret_cast<const uint8_t*>(b64_string.data());
+      const auto size = b64_string.size();
+
+      if (size == 0)
+        return {};
+
+      // Initialise the encode context
+      auto ctx = EVP_ENCODE_CTX_new();
+      EVP_DecodeInit(ctx);
+      int encoded_len = 0;
+
+      // Calculate the output buffer size: b64 is 6 bits per byte
+      int max_size = EVP_DECODE_LENGTH(size);
+      unsigned char output[max_size];
+      memset(output, '\0', max_size);
+
+      // Decode
+      int chunk_len = 0;
+      int rc = EVP_DecodeUpdate(ctx, output, &chunk_len, data, size);
+      if (rc < 0)
+      {
+        auto err = ERR_get_error();
+        char err_str[256];
```

Comment:
I would be happier if this was OPENSSL_MAX_ERROR_MSG_SIZE or something like that.

Reply:
Good point. The docs proposed 256, but using the right macro is definitely better.